StortSpelProjekt/Game/src/Gamefiles/Components/MeleeComponent.cpp

```diff
@@ -0,0 +1,174 @@
+#include "Headers/stdafx.h"
+#include "ECS/Entity.h"
+#include "MeleeComponent.h"
+#include "HealthComponent.h"
+#include "../Renderer/BoundingBoxPool.h"
+
+
+component::MeleeComponent::MeleeComponent(Entity* parent) : Component(parent)
+{
+	m_Attacking = false;
+	m_Cooldown = false;
+	m_AttackIntervall = 1.0;
+	m_TimeSinceLastAttackCheck = 0;
+	m_pMesh = nullptr;
+
+	//Create bounding box for collision for melee
+	m_pBbc = parent->GetComponent<component::BoundingBoxComponent>();
+	createCornersHitbox();
+	m_TempHitbox.CreateFromPoints(m_TempHitbox, 8, m_Corners, sizeof(DirectX::XMFLOAT3));
+	m_Hitbox = m_TempHitbox;
+
+	// Fetch the player transform
+	m_pMeleeTransform = parent->GetComponent<component::TransformComponent>()->GetTransform();
+
+	//Debugging purpose
+	if (DEVELOPERMODE_DRAWBOUNDINGBOX)
+	{
+		createDrawnHitbox(m_pBbc);
+	}
+}
+
+component::MeleeComponent::~MeleeComponent()
+{
+}
+
+void component::MeleeComponent::Update(double dt)
+{
+	m_MeleeTransformTwo = *m_pMeleeTransform;
+	float positonX = m_MeleeTransformTwo.GetPositionFloat3().x + 2*m_MeleeTransformTwo.GetRotMatrix().r[2].m128_f32[0];
+	float positonY = m_MeleeTransformTwo.GetPositionFloat3().y + 2*m_MeleeTransformTwo.GetRotMatrix().r[2].m128_f32[1];
+	float positonZ = m_MeleeTransformTwo.GetPositionFloat3().z + 2*m_MeleeTransformTwo.GetRotMatrix().r[2].m128_f32[2];
+	
+	m_MeleeTransformTwo.SetPosition(positonX, positonY, positonZ);
+	m_MeleeTransformTwo.Move(dt);
+	m_MeleeTransformTwo.UpdateWorldMatrix();
+
+	DirectX::BoundingOrientedBox temp;
+	temp = m_TempHitbox;
+
+	m_TimeSinceLastAttackCheck += dt;
+	if (m_TimeSinceLastAttackCheck > m_AttackIntervall)
+	{
+		if (m_Attacking == true)
+		{
+			Log::Print("Attack off cooldown \n");
+			m_Attacking = false;
+			m_Cooldown = false;
+		}
+		m_TimeSinceLastAttackCheck = 0;
+	}
+	temp.Transform(temp, *m_MeleeTransformTwo.GetWorldMatrix());
+	m_Hitbox = temp;
+
+}
+
+void component::MeleeComponent::Attack(bool attack)
+{
+	if (!m_Cooldown)
+	{
+		Log::Print("Attacking now \n");
+		m_Attacking = attack;
+		CheckCollision();
+		m_Cooldown = true;
+	}
+}
+
+void component::MeleeComponent::setAttackIntervall(float intervall)
+{
+	m_AttackIntervall = intervall;
+}
+
+void component::MeleeComponent::CheckCollision()
+{
+	std::vector<Entity*> list = Physics::GetInstance().SpecificCollisionCheck(&m_Hitbox);
+	if (list.size() != 0)
```

Comment:
This "if" is unneccessary. Since if the size is 0 the for loop will not run anyway.